corporate/tests/test_stripe.py

```diff
@@ -3662,7 +3667,18 @@ def create_realm(
             num_invoices: Optional[int] = None,
         ) -> Tuple[Realm, Optional[Customer], Optional[CustomerPlan], List[stripe.Invoice]]:
             realm_string_id = "realm_" + str(random.randrange(1, 1000000))
-            realm = Realm.objects.create(string_id=realm_string_id)
+            with transaction.atomic(durable=True):
+                realm = Realm(string_id=realm_string_id)
+                # For now a dummy value of -1 is given to groups fields which
+                # is changed later before the transaction is committed.
+                for (
+                    setting_group_id_name
+                ) in Realm.REALM_PERMISSION_GROUP_SETTING_ID_TO_GROUP_SETTING:
+                    setattr(realm, setting_group_id_name, -1)
+                realm.save()
+                create_system_user_groups_for_realm(realm)
+                set_default_for_realm_permission_group_settings(realm)
```

Comment:
I think a better change here would be to add a prep commit that makes this test call `do_create_realm` properly rather than trying to make a realm object hackily this other way.